### ISSUE TITLE
Exporting the svg as image does not work in Chrome, when the svg contains an embedded image

### ISSUE THREAD

**Issue by ebay-amahna** [exupero/saveSvgAsPng#58]
The functionality works as expected for an svg containing an image element in FireFox but does not work in Chrome. There are no errors in the console when executing the script, but the download dialog box does not show up in Chrome. Any clues as to what the issue could be here ? 

Thanks

**Comment by exupero** [comment]
Thanks for reporting this. Please provide an example SVG that has the problem so I can investigate.

**Comment by ebay-amahna** [comment]
<?xml version="1.0" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">

<html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=utf-8"/>
<style type="text/css">
  #line{
    width: 700px;
    margin: 20px 0;
    height: 300px;
    background: #eee;
  }
  button {
    margin: 20px 0 0 20px;
  }
​</style>
</head>
<body>
  <script type="text/javascript" src="http://mbostock.github.com/d3/d3.js?2.5.0"></script>
  <script type="text/javascript" src="saveSvgAsPng.js"></script>
  
  <a id="download" href="#">Download SVG</button>
  <div id="line">
      <svg width="810" height="3109">
<g transform="translate(0,0)"><g>
<image xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="1066319783.html.jpg" width="810" height="3109" x="0" y="0"></image>
</g>
<rect x="98" y="160" id="idlabel_0" opacity="0.65" stroke="black" stroke-width="2" style="fill:  rgb(255, 0, 0);" width="24.34375" height="18.4375"></rect>
<rect x="441" y="-5" id="idlabel_1" opacity="0.65" stroke="black" stroke-width="2" style="fill: rgb(0, 204, 255);" width="18.21875" height="18.4375"></rect>
<rect x="582" y="1549" id="idlabel_2" opacity="0.65" stroke="black" stroke-width="2" style="fill: rgb(0, 204, 255);" width="18.21875" height="18.4375"></rect>
<rect x="78" y="40" id="idlabel_3" opacity="0.65" stroke="black" stroke-width="2" style="fill: rgb(0, 204, 255);" width="18.21875375" height="18.4375"></rect>
<text x="100" y="160" font-family="sans-serif" font-weight="bold" font-size="11" fill="white" dy="1.5em" text-anchor="start">273</text>
<text x="443" y="-5" font-family="sans-serif" font-weight="bold" font-size="11" fill="white" dy="1.5em" text-anchor="start">70</text>
<text x="80" y="122" font-family="sans-serif" font-weight="bold" font-size="11" fill="white" dy="1.5em" text-anchor="start">17</text>
<text x="100" y="650" font-family="sans-serif" font-weight="bold" font-size="11" fill="white" dy="1.5em" text-anchor="start">17</text>
<text x="617" y="352" font-family="sans-serif" font-weight="bold" font-size="11" fill="white" dy="1.5em" text-anchor="start">5</text></g></svg>
  </div>​
  
   <script type="text/javascript">
      
    d3.select("#download").on("click", function(){
        saveSvgAsPng(document.getElementsByTagName("svg")[0], "svg-snapshot.png");	
        
    })
   </script>
</body>
</html>

**Comment by ebay-amahna** [comment]
[svg-print.txt](https://github.com/exupero/saveSvgAsPng/files/58965/svg-print.txt)

**Comment by ebay-amahna** [comment]
The svg contains an inline image and attempt at png conversion is throwing this error:

```
saveSvgAsPng.js:34 Uncaught SecurityError: Failed to execute 'toDataURL' on
'HTMLCanvasElement': Tainted canvases may not be exported.
img.onload @ saveSvgAsPng.js:34
```

This works fine in FF though.

**Comment by ebay-amahna** [comment]
FYI....this issue with Chrome loading CORS image to a canvas is documented here : https://developer.mozilla.org/en-US/docs/Web/HTML/CORS_enabled_image. 

I'm getting the issue despite the fact that the image is loaded from the same domain and has the
Access-Control-Allow-Origin header set to '*'.

**Comment by vinoth-rz** [comment]
I am also facing the same problem, below is my image data in svg

```
<image
xlink:href="data:img/png;base64,iVBORw0KGgoAAAANSUhEUgAAAFAAAAA/CAMAAABAbb8QAAAABGdBTUEAALGPC/xhBQAAACBjSFJNAAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAAilBMVEUAAAAAAAAAAACnKRkxDAcAAADyPCTyPCTyPCTyPCTyPCTyPCTyPCTyPCTyPCTyPCTyPCTyPCTyPCTyPCQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAC8LxxhGA4AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADyPCQeCAXe6RXhAAAAK3RSTlMAYMDonoDAQNBg4IAQ8JAgoHAwsBBoWEAoIIwI96BI4DB4GHA4UNy8wrDUn6DeAQAAAAFiS0dEAIgFHUgAAAAJcEhZcwAACxIAAAsSAdLdfvwAAAGpSURBVFjDrdbpVoMwEAXgAaVaq9YFKDTasroR3//1DDFVKJBMkpl/7en5yrkMlwAYJwjDi0vzz/ATdR3nqytikPPrNTHI+c3mlhbk/O5+SwuKefAPcwyK++Mb5jkowlx7hTkFRZgbjzDnwD5MZ3IBdF/2RbAPkxh0W3YtKO6PdZgG0H7ZzaDlsmNAEeYjOkwcaLHsWLAP84kY5CvMReJB3AViQXxfoECLm4wB7TrXCD5bto4etH+UtaDTm3UZdKvDRdD5dDIPur9S5kCvl94U9Hwtn4PeB4cxSHC0GYA0h68/ELnEcRKJSQI16e40Wa5mL0H8AZalkX5CAb68jv4215MHI/h9HHxOC2a6yKI0gN3RQuunqlFg0qA0GWRiBJMmtlqOVgu+WWoySB0YWnN9kLO3pj7sXcGZIMs2k0+KIwisnWjgBQI0/+ubnb7zAiGXQZbD9fUDofotCzpQlUVZkYEAOxliQQdCJoNs6UBVFgEjA4EF8impyEDVumVBB6qyONCBqiwCRgaqsqjd22Y6sizeCUEZpHPBzo4oix78IAMhTj8F+IX+/Q8BvrYsscQ/JQAAAABJRU5ErkJggg==" />
```

I am trying to convert it to png, but the image tag elements are neglected in image

**Comment by exupero** [comment]
@vinoth-rz Try changing the mimetype in your data URI to `image/png`.

**Comment by exupero** [comment]
@ebay-amahna Your example shouldn't have a problem with CORS because it isn't loading the image using an absolute path.

Does it work if you convert your image from a JPG to a PNG?

**Comment by ebay-amahna** [comment]
@exupero, I was able to find the reason why the solution didn't work in Chrome for my use-case. It took a day to identify that as In Chrome, there was no error on the console, but the svg wouldn't download. 

I was hitting a size limit on the canvas. Using a scaling factor of .6 resolved my issue. I will figure out the exact size limit so that the scaling can be added conditionally.

**Comment by exupero** [comment]
I'm glad you figured it out! Can you share what references you found about the size limit? You're right that it would be good to warn users when the image will be too big.

**Comment by ebay-amahna** [comment]
I wanted to update you on this issue. Based on information in this link (http://stackoverflow.com/questions/6081483/maximum-size-of-a-canvas-element), I realized that I seemed to be hitting the issue with much smaller images e.g. the specific image that wouldn't work for me in Chrome had a width of 800 px and height of 3100 px.  I modified the code canvas.toDataUrl was getting invoked to use 'image/jpeg' instead of 'image/png' and the code worked like a charm after that with no scaling needed. I was able to test with the largest possible image for our app and still didn't run into issues.
